yt_dlp/extractor/amcnetworks.py

```diff
@@ -65,15 +65,36 @@ class AMCNetworksIE(ThePlatformIE):
     def _real_extract(self, url):
         site, display_id = re.match(self._VALID_URL, url).groups()
         requestor_id = self._REQUESTOR_ID_MAP[site]
-        properties = self._download_json(
-            'https://content-delivery-gw.svc.ds.amcn.com/api/v2/content/amcn/%s/url/%s' % (requestor_id.lower(), display_id),
-            display_id)['data']['properties']
+        page_data = self._download_json(
+            'https://content-delivery-gw.svc.ds.amcn.com/api/v2/content/amcn/%s/url/%s'
+            % (requestor_id.lower(), display_id), display_id)['data']
+        properties = page_data.get('properties')
         query = {
             'mbr': 'true',
             'manifest': 'm3u',
         }
-        tp_path = 'M_UwQC/media/' + properties['videoPid']
-        media_url = 'https://link.theplatform.com/s/' + tp_path
+
+        has_releasePid = False
+        video_player_count = 0
+        try:
+            for v in page_data['children']:
+                if v['type'] == 'video-player':
+                    releasePid = \
+                        v['properties']['currentVideo']['meta']['releasePid']
+                    tp_path = 'M_UwQC/' + releasePid
+                    media_url = 'https://link.theplatform.com/s/' + tp_path
+                    has_releasePid = True
+                    video_player_count += 1
+        except KeyError:
+            pass
+        if video_player_count > 1:
+            self.report_warning('The JSON data has more than one video player.')
```

Comment:
Does the various "players" contain different formats? or are they all effectively the same?

Reply:
I haven't encountered a web page (for that extractor) with more than one embedded player. The JSON data theoretically could have more than one 'type':'video-player' key/value pair, however, which is why the warning message is there.